frontend/public/components/modals/cluster-update-modal.tsx

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable no-undef, no-unused-vars */
+
+import * as _ from 'lodash-es';
+import * as React from 'react';
+
+import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter, ModalComponentProps } from '../factory/modal';
+import { Dropdown, PromiseComponent } from '../utils';
+import { k8sPatch, K8sResourceKind } from '../../module/k8s';
+
+class ClusterUpdateModal extends PromiseComponent {
+  public state: ClusterUpdateModalState;
```

Comment:
```suggestion
  readonly state: ClusterUpdateModalState = {
    selectedVersion: 0,
  };
```

Reply:
`PromisComponent` also initializes state, wouldn't this override that?

Reply:
Ah, OK. Disregard.

We should start trying to get away from inheritance in these components.